### PR TITLE
Makes t-rays fast processing

### PR DESCRIPTION
**Pull request by silicons**
It's kinda painful running around with the slow objects processing rate, as you have to stand still to scan things :/
:cl:
tweak: T-ray scanners are now much faster at scanning.
/:cl:

### CODE REVIEW

**Comment by WJohn** [comment]
i dont actually know if this is more or less intensive but whatever

**Comment by RemieRichards** [comment]
This is definitely more intensive, can you really not tell that little WJohn?

**Comment by WJohn** [comment]
no of course not i'm completely blind to code

**Comment by RemieRichards** [comment]
Yes, but the basic principle of "if this thing happens more often it's more costly" is universal.
If more work occurs, it takes more work! that's <i>obvious.</i>

**Comment by RandomMarine** [comment]
Question is if it's a _significant_ increase. Would the server slow to a crawl when a bunch of people are running around with active T-ray scanners?

**Comment by RemieRichards** [comment]
Every ```process()``` it has to update the invisibility of everything under the ground in a 2 tile radius, and create an image for it, and then show that image to everyone within a 7 tile radius, then do it all again for every other ```process()``` as long as it's still on.

It certainly does NOT help that t-rays use a ```spawn(10)``` to delay things by a second, and a spawn is about as costly as 20-30 proc calls.

tl;dr It's not hitler's groin worth of cost, but it's most certainly not free.

**Comment by Arianya** [comment]
Out of interest, why delay t-rays by a second? Its not like they give exciting/balance affecting information that is made nicer by a second delay, and if its as expensive as 20-30 proc calls...

**Comment by RemieRichards** [comment]
@Arianya Ironically, the delaying by a second (using the expensive spawn) was probably for performance!

**Comment by XDTM** [comment]
No, it actually made the pipe un-invisible then turned it back after the spawn(10).

**Comment by RemieRichards** [comment]
@XDTM Read what I said, nowhere did I say the image was OF the pipe (it's not, it's an effect)

**Comment by XDTM** [comment]
The image is indeed just a circular pulse, i noticed while refactoring it; the pipe image is from the pipe becoming visible. Still, the spawn is not used to delay the effect for performance, but just to return the pipe to normal.

**Comment by kevinz000** [comment]
I'll look into another solution later this was mostly so people can actually track mobs with these.
